scripts/lib/CIME/nmlgen.py

```diff
@@ -707,33 +729,61 @@ def _write_input_files(self, data_list_path):
+                                                              file_path=file_path,
+                                                              input_pathname=input_pathname,
+                                                              lines_hash=lines_hash)
+
+    def _add_file_to_input_data_list(self, input_data_list, variable_name, file_path, input_pathname, lines_hash):
+        """Add one file to the input data list, if needed
+
+        It's possible that file_path actually contains multiple files delimited by
+        GRID_SEP. (This is the case when a component has multiple grids, and so has a file
+        for each grid.) In this case, we split it on GRID_SEP and handle each separated
+        portion as a separate file.
+
+        Args:
+        - input_data_list: file handle
+        - variable_name (string): name of variable to add
+        - file_path (string): path to file
+        - input_pathname (string): whether this is an absolute or relative path
+        - lines_hash (set): set of hashes of lines already in the given input data list
+
+        """
+        for one_file_path in file_path.split(GRID_SEP):
+            # NOTE - these are hard-coded here and a better way is to make these extensible
+            if one_file_path == 'UNSET' or one_file_path == 'idmap' or one_file_path == 'idmap_ignore':
+                continue
+            if input_pathname == 'abs':
+                # No further mangling needed for absolute paths.
+                # At this point, there are overwrites that should be ignored
+                if not os.path.isabs(one_file_path):
+                    continue
+                else:
+                    pass
+            elif input_pathname.startswith('rel:'):
+                # The part past "rel" is the name of a variable that
+                # this variable specifies its path relative to.
+                root_var = input_pathname[4:]
+                root_dir = self.get_value(root_var)
+                one_file_path = os.path.join(root_dir, one_file_path)
+            else:
+                expect(False,
+                       "Bad input_pathname value: {}.".format(input_pathname))
+
+            # Write to the input data list.
+            #
+            # Note that the same variable name is repeated for each file. This currently
+            # seems okay for check_input_data, but if it becomes a problem, we could
+            # change this, e.g., appending an index to the end of variable_name.
+            string = "{} = {}".format(variable_name, one_file_path)
+            hashValue = hashlib.md5(string.rstrip().encode('utf-8')).hexdigest()
+            if hashValue not in lines_hash:
+                logger.debug("Adding line {} with hash {}".format(string,hashValue))
+                input_data_list.write(string+"\n")
+            else:
+                logger.debug("Line already in file {}".format(string))
```

Comment:
The body of this for loop is essentially the same as the pre-existing code, except `file_path` is replaced with `one_file_path`